UnityProject/Assets/Scripts/Items/Food/Torch.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using Mirror;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Random = UnityEngine.Random;
+using Items;
+
+namespace items
+{
+    /// <summary>
+    /// Lightable objects that can be used as lighters
+    /// </summary>
+    public class Torch : NetworkBehaviour, ICheckedInteractable<HandApply>,
+        ICheckedInteractable<InventoryApply>, IServerDespawn
+    {
+		private const int DEFAULT_SPRITE = 0;
+		private const int LIT_SPRITE = 1;
+
+		[SerializeField]
+		[Tooltip("Object to spawn after torch burnt")]
+		private GameObject burntOutPrefab = null;
+
+		[SerializeField]
+		[Tooltip("Time after torch will destroy and spawn burnt remains")]
+		private float burnTimeSeconds = 30;
+
+		public SpriteHandler spriteHandler;
```

Comment:
make it private and get the spriteHandler in ``Awake`` unless the prefab that will use it has more than one ``SpriteHandler``